src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedLocal.cs

```diff
@@ -79,6 +82,19 @@ public override RefKind RefKind
             get { return _refKind; }
         }
 
+        internal override bool IsWritableVariable
+        {
+            get
+            {
+                if (SynthesizedKind == SynthesizedLocalKind.AsyncIteratorCancellationToken)
+                {
+                    return false;
+                }
+
+                return base.IsWritableVariable;
```

Comment:
Why is `_refKind` not considered here? Consider that `in` would not be writable.